collector/collector.go

```diff
@@ -156,6 +158,11 @@ func (c *collector) waitForStartup(ctx context.Context, startupErr chan error) e
 			c.svc.Shutdown()
 			return ctx.Err()
 		case err := <-startupErr:
+			if err == nil {
```

Comment:
@BinaryFissionGames i'm obviously missing something as i'm trying to wrap my head around these functions and changes. doesn't this just ensure that we return an error every single time run is called now?

Reply:
It doesn't always return an error; This function has 3 exit points. It can exit if:
1. The startupErr channel is signalled. In this case, we attempted to run the collector, but the collector is no longer running; So the collector failed to start. We should always return a non-nil error in this case.
2. The context is cancelled (returns ctx error).
3. The service state is `Running` - in this case, no error is returned (https://github.com/observIQ/observiq-otel-collector/blob/129f7b1e36114eaf014dfea2e5698506d05ae9ae/collector/collector.go#L150)